### DISCUSSION TITLE
error mount volume  volume target must be an absolute path

### DISCUSSION THREAD

**Discussion by hdez-es** [containerd/nerdctl#2340] [Q&A]
### Description

I want to create a MariaDB container using nerdctl and a compose file, but it shows this error:
C:\containers>nerdctl compose -f test.yml up
level=fatal msg="volume target must be an absolute path, got \"/var/lib/mysql\""

OS: Windows server 2022 with WSL2
nerdctl version 1.4.0

compose file: test.yml

version: '3.8'

services:

  mariadb:
    image: mariadb
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: passwroot
      MYSQL_DATABASE: db
      MYSQL_USER: user
      MYSQL_PASSWORD: pass
    volumes:
      - mariadb-data:/var/lib/mysql

volumes:
  mariadb-data:

### Steps to reproduce the issue

1.
2.
3.

### Describe the results you received and expected

C:\containers>nerdctl compose -f test.yml up
level=fatal msg="volume target must be an absolute path, got \"/var/lib/mysql\""

### What version of nerdctl are you using?

nerdctl version 1.4.0

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information

Windows server 2022 WSL2

**Comment by AkihiroSuda** [comment]
Are you seeing this with Rancher Desktop for Windows, or with bare WSL2?

**Comment by hdez-es** [comment]
I've installed the containerd service in windows using this guide
https://github.com/containerd/containerd/blob/main/docs/getting-started.md section "Installing containerd on Windows"

Then I run nerdctl.exe version 1.4.0
nerdctl compose -f test.yml up 

The error is level=fatal msg="volume target must be an absolute path, got "/var/lib/mysql""

compose file: test.yml

version: '3.8'

services:

mariadb:
image: mariadb
restart: always
environment:
MYSQL_ROOT_PASSWORD: passwroot
MYSQL_DATABASE: db
MYSQL_USER: user
MYSQL_PASSWORD: pass
volumes:
- mariadb-data:/var/lib/mysql

volumes:
mariadb-data:

**Comment by AkihiroSuda** [comment]
Please make sure that you are running the Linux version of containerd and nerdctl for running Linux containers
